### PR TITLE
WIP: Chore/license compliance

### PR DESCRIPTION
**Pull request by bufferoverflow**
This is related to #83 , it adds repolinter (from my branch until https://github.com/todogroup/repolinter/pull/63 is merged) and the license header as suggested by reuse.software. I will add all those headers if you @fh1ch agree on this.

### CODE REVIEW

**Comment by bufferoverflow** [comment]
@fh1ch can we drop support for node 4 and 5 ?

File: LICENSE.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 The MIT License (MIT)
 
-Copyright (c) 2017 fh1ch
+Copyright (c) 2017 Fabio Huser <fabio@fh1.ch>
```
Comment (fh1ch) [inline LICENSE.md]: Good point 👍 

The`README.md` already uses the correct copyright information (see https://github.com/fh1ch/node-bacstack/blob/master/README.md#license)

File: test/unit/utils.js [hunk 2]
```diff
@@ -1,3 +1,5 @@
+// Copyright (C) 2017 Fabio Huser <fabio@fh1.ch>
+// SPDX-License-Identifier: MIT
```
Comment (fh1ch) [inline test/unit/utils.js]: See https://github.com/fh1ch/node-bacstack/issues/83#issuecomment-354143543

File: package.json [hunk 3]
```diff
@@ -39,7 +40,8 @@
   "homepage": "https://github.com/fh1ch/node-bacstack#readme",
   "dependencies": {
     "debug": "^3.1.0",
-    "iconv-lite": "^0.4.19"
+    "iconv-lite": "^0.4.19",
+    "repolinter": "bufferoverflow/repolinter#fix/reuse-software"
```
Comment (fh1ch) [inline package.json]: We shouldn't merge something which relies on NPM packages on GitHub/special branches.

**Comment by fh1ch** [comment]
> @fh1ch can we drop support for node 4 and 5 ?

Well, not really. Node 4 LTS is still maintained and used especially on embedded devices (as they are a bit hard to update). Also, a lot of effort was invested to keep it compatible with Node 4 LTS, which I don't want to loose. I also don't really see the point of introducing repolinter, since most of the tasks are covered by other tooling (e.g. https://github.com/fh1ch/node-bacstack/community).

File: package.json (hunk 3 above)
Comment (bufferoverflow) [reply package.json]: bingo, that's why this is WIP

File: LICENSE.md (hunk 1 above)
Comment (bufferoverflow) [reply LICENSE.md]: yep, I know

**Comment by bufferoverflow** [comment]
I just had a look at https://github.com/imyller/meta-nodejs/tree/master/recipes-devtools/nodejs and I would say it's not an issue for embedded devices to use node > 5. An embedded system using node should be able to update anytime, otherwise node would be the wrong stack anyway.

**Comment by bufferoverflow** [comment]
https://github.com/fh1ch/node-bacstack/community is nice, but repolinter goes beyond and is independent from the hosting provider.

**Comment by fh1ch** [comment]
- **Alignment of copyright claim:** Since the new year is approaching us today and the project needs to bump the license year anyway, I've created #87 which not only updates the year, but also aligns the copyright claims.
- **Introduction of copyright / license headers:** The introduction of headers have been declined by the majority of maintainers as part of #83 and therefore won't be accepted.
- **Introduction of repolinter:** As already partially addressed in #83, I personally doubt the value of adding repolinter. The checks are redundant if considering GitHub's Community Analytics and the Core Infrastructure Best Practises, while the adaption of repolinter in the wild is quiet poor (e.g. same amount of stars as this repo, despite the fact that the ToDo group has more than 16 members). Additionally, the fact that it's not compliant with Node 4&5, while even our ES6 integration (see #84) honoured it, questions the value of it. On the other hand, I've just split the Travis-CI builds into stages, which would support to run certain tasks with one specific version of Node.

What's your thought of this?

**Comment by bufferoverflow** [comment]
I do not agree, but I do not have some interest to discuss further. It‘s your own project and you will anyway do what you like. It was impressive to see that you created headerless.org to have a reference for that approach for your project.

**Pull request closed without merging**